Makefile

```diff
@@ -1,83 +1,183 @@
-    if [ -n "${TESTDIR}" ]; then \
-        if [ "${TESTDIR}" = "upgradecluster" ]; then \
-            go test -timeout=45m -v ./upgradecluster/... -installVersionOrCommit "${INSTALLTYPE}"; \
-        elif [ "${TESTDIR}" = "versionbump" ]; then \
-            go test -timeout=45m -v -tags=versionbump ./versionbump/... -cmd "${CMD}" -expectedValue "${VALUE}" \
-            -expectedValueUpgrade "${VALUEUPGRADED}" -installVersionOrCommit "${INSTALLTYPE}" -channel "${CHANNEL}" -testCase "${TESTCASE}" \
-            -deployWorkload "${DEPLOYWORKLOAD}" -workloadName "${WORKLOADNAME}" -description "${DESCRIPTION}"; \
-		elif [ "${TESTDIR}" = "mixedoscluster" ]; then \
-            go test -timeout=45m -v -tags=mixedos ./mixedoscluster/...; \
-        fi; \
-    elif [ -z "${TESTDIR}" ]; then \
-        go test -timeout=45m -v ./createcluster/...; \
-    fi;'
+	if [ -z "$${IMGNAME}" ]; then IMGNAME=${IMGNAME}; fi; \
+	if [ -z "$${TAGNAME}" ]; then TAGNAME=${TAGNAME}; fi; \
+	  docker run -dt --name acceptance-test-$${IMGNAME} \
+	  -e AWS_ACCESS_KEY_ID=$${AWS_ACCESS_KEY_ID} \
+	  -e AWS_SECRET_ACCESS_KEY=$${AWS_SECRET_ACCESS_KEY} \
+	  --env-file ./config/.env \
+	  -v ${ACCESS_KEY_LOCAL}:/go/src/github.com/rancher/distros-test-framework/config/.ssh/aws_key.pem \
+	  -v ./scripts/test-runner.sh:/go/src/github.com/rancher/distros-test-framework/scripts/test-runner.sh \
+	  acceptance-test-${TAGNAME}
+
+test-run-state:
+	DOCKERCOMMIT=$$? \
+	CONTAINER_ID=$(shell docker ps -a -q --filter name=acceptance-test-${IMGNAME}); \
+    	if [ -z "$$CONTAINER_ID" ]; then \
+    		echo "No matching container found."; \
+    		exit 1; \
+    	else \
+    		docker commit $$CONTAINER_ID teststate:latest; \
+    		if [ $$DOCKERCOMMIT -eq 0 ]; then \
+    			docker run -dt --name "${TESTSTATE}" --env-file ./config/.env \
+    			-e AWS_ACCESS_KEY_ID=$${AWS_ACCESS_KEY_ID} \
+    			-e AWS_SECRET_ACCESS_KEY=$${AWS_SECRET_ACCESS_KEY} \
+    			-v $${ACCESS_KEY_LOCAL}:/go/src/github.com/rancher/distros-test-framework/config/.ssh/aws_key.pem \
+    			-v ./scripts/test-runner.sh:/go/src/github.com/rancher/distros-test-framework/scripts/test-runner.sh \
+    			teststate:latest \
+    			echo "Docker run exit code: $$?"; \
+    		else \
+    			echo "Failed to commit container"; \
+    			exit 1; \
+    		fi; \
+    	fi
+
+## USE THIS TO RUN TESTS IN A TOTALLY FRESH ENVIRONMENT IN DOCKER
+.PHONY: test-complete
+test-complete: test-env-clean test-env-down remove-tf-state test-env-up test-run
+
 
 .PHONY: test-logs
 test-logs:
 	@docker logs -f acceptance-test${IMGNAME}
 
 .PHONY: test-env-down
 test-env-down:
-	@echo "Removing containers and images"
-	@docker stop $$(docker ps -a -q --filter="name=acceptance-test*")
-	@docker rm $$(docker ps -a -q --filter="name=acceptance-test*") ; \
-	 docker rmi --force $$(docker images -q --filter="reference=acceptance-test*")
+	@echo "Removing containers"
+	@docker ps -a -q --filter="name=acceptance-test*" | xargs -r docker rm -f || true
```

Comment:
Can we add a check to not error out if the container does not exists?

Reply:
sure